### PR TITLE
Chembombs 2

### PR DESCRIPTION
**Pull request by nullbear**
Fixes issue with https://github.com/tgstation/-tg-station/pull/16475, allowing it to be merged. Easier and faster to do it this way.

### CODE REVIEW

File: code/game/machinery/syndicatebomb.dm [hunk 1]
```diff
@@ -292,6 +342,128 @@
...
+			for(var/atom/A in T.GetAllContents())
+				if(A in mycontents) continue
+				if(!(A in viewable)) continue
+				reactable |= A
+		if(!reactable.len) //Nothing to react with. Probably means we're in nullspace.
+			qdel(src)
+			return
+		var/fraction = 1/reactable.len
+		for(var/atom/A in reactable)
+			reagents.reaction(A, TOUCH, fraction)
```
Comment (phil235) [inline code/game/machinery/syndicatebomb.dm]: is this part copypasted from grenade code?

File: code/game/machinery/syndicatebomb.dm [hunk 2]
```diff
@@ -292,6 +342,128 @@
...
+				if(!NT.CanAtmosPass(T)) continue
+				reachable |= T
+				break
+	return reachable
+
+/obj/item/weapon/bombcore/chemical/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/crowbar) && beakers.len > 0)
+		for (var/obj/item/B in beakers)
+			B.loc = get_turf(src)
+			beakers -= B
```
Comment (phil235) [inline code/game/machinery/syndicatebomb.dm]: no sound when crowbaring?

File: code/game/machinery/syndicatebomb.dm (hunk 1 above)
Comment (nullbear) [inline code/game/machinery/syndicatebomb.dm]: the alternative is hacky, or to implement global chem_mixing procs and change grenades to use those instead. This is faster and more efficient than either.

File: code/game/machinery/syndicatebomb.dm (hunk 2 above)
Comment (nullbear) [inline code/game/machinery/syndicatebomb.dm]: this is actually supposed to be "attack_self". forgot to change it.

**Comment by nullbear** [comment]
better?

**Comment by nullbear** [comment]
@KorPhaeron Anything that needs to be done with this, to have it merged?

**Comment by KorPhaeron** [comment]
Get someone who knows how to code to review it would be the most important part

**Comment by nullbear** [comment]
@phil235 You're pretty good at calling out my laziness. You wanna review this?

**Comment by GunHog** [comment]
Just run your final code locally, make sure everything works (I cannot use the "t" word), and fix anything that is broken.

File: code/modules/research/designs/weapon_designs.dm [hunk 3]
```diff
@@ -127,6 +127,42 @@
...
+	req_tech = list("combat" = 3, "materials" = 3)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000, MAT_SILVER = 500)
+	reliability = 79
+	build_path = /obj/item/weapon/grenade/chem_grenade/cryo
+	category = list("Weapons")
+*/
+
+/datum/design/adv_grenade
+	name = "Advanced Release Grenade"
```
Comment (GunHog) [inline code/modules/research/designs/weapon_designs.dm]: A....LIGHT GRENADE?

File: code/modules/research/designs/weapon_designs.dm [hunk 4]
```diff
@@ -127,6 +127,42 @@
 	build_path = /obj/item/weapon/grenade/chem_grenade/large
 	category = list("Weapons")
 
+/datum/design/pyro_grenade
+	name = "Pyro Grenade"
+	desc = "An advanced grenade that is able to self ignite its mixture."
```
Comment (GunHog) [inline code/modules/research/designs/weapon_designs.dm]: This thing will be **GOOFTASTIC**

File: code/game/atoms.dm [hunk 5]
```diff
@@ -1,404 +1,404 @@
...
-			clear_reagents_to_vomit_pool(M,V)
-
-/atom/proc/clear_reagents_to_vomit_pool(mob/living/carbon/M, obj/effect/decal/cleanable/vomit/V)
-	M.reagents.trans_to(V, M.reagents.total_volume / 10)
-	for(var/datum/reagent/R in M.reagents.reagent_list)                //clears the stomach of anything that might be digested as food
-		if(istype(R, /datum/reagent/consumable))
-			var/datum/reagent/consumable/nutri_check = R
-			if(nutri_check.nutriment_factor >0)
-				M.reagents.remove_reagent(R.id,R.volume)
+/atom
```
Comment (GunHog) [inline code/game/atoms.dm]: Try to find out what made Github think you removed and readded all these lines.

Comment (nullbear) [inline code/game/atoms.dm]: This was the merge update. So someone else did this.

File: code/modules/research/designs/weapon_designs.dm (hunk 3 above)
Comment (nullbear) [inline code/modules/research/designs/weapon_designs.dm]: The cryo im thinking would be for stuff like freezing wet floors into ice, or frostoil stuff, etc. idk man. It's just intended as a compliment to the pyro.

I tested a flashbang 'advanced release bomb' with 4 bluespace beakers, and it's a pretty effective area denial device. If you enter the radius, you're flashed until you go blind.

File: code/game/atoms.dm (hunk 5 above)
Comment (nullbear) [inline code/game/atoms.dm]: Yeah, looking at it. What happened is i did an update from master. And didn't look through it really. I'd have had to look through it pretty meticulously to notice this, and then discard the changes. Nothing is changed though, so it wont be an issue.

Also accidently didn't include a couple files as part of the update from master. Included those now.

**Comment by nullbear** [comment]
atoms.dm file is just github being dumb. Won't have any impact, since nothing is changed.

**Comment by nullbear** [comment]
okay i think i figured it out. i'm not used to using the shell.

**Pull request closed without merging**